Automattic/jetpack
# AI Assistant: Connect usage data to UsagePanel

Part of #33359.

## Proposed changes:

* Connect the usage bar on the usage panel to real usage data, so we display the current usage for the site
* Hide the `Requests will reset in X days.` message since the current plans do not reset; I kept the message there because it will be necessary for the tiered plans
* For paid plans, since there is not real limit, I am showing `X / ∞ requests` and fixing the usage percentage to 10%  so the usage bar shows some information

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Test on a site with beta extensions enabled
* Go to the block editor and open the Jetpack sidebar; look for the "AI Assistant" section:

<img width="323" alt="Screenshot 2023-10-19 at 18 05 48" src="https://github.com/Automattic/jetpack/assets/6760046/96564e80-dc8a-4dca-92be-de071c09d9ce">

---

* On the "AI Assistant" section, confirm you see the "Usage" panel with a message stating your usage + the usage bar:

<img width="269" alt="Screenshot 2023-10-20 at 16 04 15" src="https://github.com/Automattic/jetpack/assets/6760046/b44ed481-123e-45d5-b688-1d46b158851a">

---

* To confirm the number is correct, open the WordPress.com Console (https://developer.wordpress.com/docs/api/console/) and do a request to `wpcom/v2` `/sites/YOUR-SITE/jetpack-ai/ai-assistant-feature`:

<img width="600" alt="Screenshot 2023-10-20 at 16 09 32" src="https://github.com/Automattic/jetpack/assets/6760046/d4bd2c2a-c7bf-461c-9ec3-2d0878b40c5c">

---

* Test it on a site with the Jetpack AI plan and confirm you see the usage as a following (X / ∞):

<img width="300" alt="Screenshot 2023-10-20 at 16 10 33" src="https://github.com/Automattic/jetpack/assets/6760046/2c795184-330f-4835-8d1b-61ed36f4fe91">

---

* Also test on a site without the plan and confirm you see the usage in another way (X / Y):

<img width="300" alt="Screenshot 2023-10-20 at 16 10 55" src="https://github.com/Automattic/jetpack/assets/6760046/ad5a8602-7a64-4057-a595-1fdf4796524f">

---

* Notice that the `Requests will reset in X days.` is not visible on both cases
